src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameFormatter.cs

```diff
@@ -99,24 +99,36 @@ private static void AppendQuoted(this StringBuilder sb, string s)
 
             for (int i = 0; i < s.Length; i++)
             {
-                bool addedEscape = false;
-                foreach (KeyValuePair<char, string> kv in EscapeSequences)
+                switch (s[i])
                 {
-                    string escapeReplacement = kv.Value;
-                    if (!(s[i] == escapeReplacement[0]))
+                    case '\\':
+                    case ',':
+                    case '=':
+                    case '\'':
+                    case '"':
+                        sb.Append('\\');
+                        break;
+                    case '\t':
+                        sb.Append("\\t");
                         continue;
-                    if ((s.Length - i) < escapeReplacement.Length)
+#if TARGET_WINDOWS
```

Comment:
This Windows specific behavior sounds like a bug introduced in the managed rewrite of AssemblyName formatters. It does not really make sense for the AssemblyName escaping rules to differ between Windows and Unix.

I would recommend to change the managed formatter to do what the unmanaged formatter has been always doing: https://github.com/dotnet/runtime/blob/master/src/coreclr/src/binder/textualidentityparser.cpp#L740

Reply:
Done.